### DISCUSSION TITLE
Extension host terminated unexpectedly

### DISCUSSION THREAD

**Discussion by Difocd** [ocamllabs/vscode-ocaml-platform#1168] [Q&A]
Type: <b>Bug</b>

1. follow install steps on ocaml official page
2. `dune init project helloworld`
3. open it using vscode

Extension version: 1.12.2
VS Code version: Code 1.78.2 (Universal) (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:44:45.204Z)
OS version: Darwin x64 19.6.0
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (8 x 2200)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (2.02GB free)|
|Process Argv|. --crash-reporter-id 828a3c02-29fa-4b94-ac07-efbc6837bd0a|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
2i9eh265:30646982
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
03d35959:30757346
pythonfmttext:30731395
pythoncmv:30756943
fixshowwlkth:30730052
pythongtdpathcf:30739705
pythonnosmt12:30757131

```

</details>

**Comment by Difocd** [comment]
Having uninstalled all other vscode extensions except this one.

**Comment by Difocd** [comment]
I have been tackling this problem with a friend for two hours. We find that there is some conflict between ocaml-platform and vscode builtin extensions. When vscode is already properly started and then ocaml-platform is installed, everything will be ok.

### We have tried:
1. Modify `activationEvents` of `package.json`. This is useless in a latest version of vscode.
2. Modify `dist/vscode_ocaml_platform.bc.js` to add a delay start. This can not get rid of crash either.

### Final workaround:
Disable the ocaml-platform extension every time closing vscode of a ocaml project and enable the extension when reopen it. This can effectively avoid crash, at least for now.

**Comment by 3Rafal** [comment]
Hello, thanks for the report. I tried to reproduce this bug, but everyting worked fine on my machine. I use macOS, with the same VSCode version but my OS kernel version is newer - I'm on  Darwin Kernel Version 21.6.0:.

Could you provide more info on:
>When vscode is already properly started and then ocaml-platform is installed, everything will be ok.

What do you mean by vscode being "properly started"? And how does it relate to installation of ocaml-platform?

**Comment by Difocd** [comment]
> What do you mean by vscode being "properly started"? And how does it relate to installation of ocaml-platform?

The first time I use ocaml-platform, I have an already started vscode, and I install ocaml-platform from market place. Everything is just fine. 

Next day, when I open vscode (**with ocaml-platform already installed and enabled**), the crash started to arise.

I suspect that ocaml-platform and vscode can not boot at the same time. So I disable ocaml-platform, restart vscode and then re-enable ocaml-platform. The crash go away, and now I can finally use ocaml-platform.

I hope these information could be helpful.

**Comment by ozanvos** [comment]
I was able to reproduce this on a fresh VSCode profile (both stable VSCode and VSCode Insiders have the issue).

In my tests the issue seems to be caused by some interaction between GitHub Copilot extension and OCaml Platform extension. To reproduce, you need to have an active GitHub Copilot subscription and have logged in with your GitHub account after installing the extension.

https://github.com/ocamllabs/vscode-ocaml-platform/assets/111265/0e48d467-967d-4282-8060-234a5b6e0113
